src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,158 @@
+                if (currentNode.key == null) {
+                    currentNode.value = value;
+                    return;
+                }
+                currentNode = currentNode.next;
+            }
+            if (currentNode.key == null) {
+                currentNode.value = value;
+            } else {
+                currentNode.next = new Node<>(null, value, null);
+                size++;
+            }
+            if (size > table.length * DEFAULT_LOAD_FACTOR) {
+                resize();
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
+        if (key == null) {
+            return getForNullKey();
+        }
+        int index = calculateBucketIndex(key);
+        Node<K, V> currentNode = table[index];
+        while (currentNode != null) {
+            if (Objects.equals(currentNode.key, key)) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
         return null;
     }
 
+    private V getForNullKey() {
+        if (table[0] != null) {
+            Node<K, V> currentNode = table[0];
+            while (currentNode != null) {
+                if (currentNode.key == null) {
+                    return currentNode.value;
+                }
+                currentNode = currentNode.next;
+            }
+        }
+        return null;
+    }
+
+    private void resize() {
+        int newCapacity = table.length << 1;
+        Node<K, V>[] oldTable = table;
+        table = new Node[newCapacity];
+        size = 0;
+        for (Node<K, V> node : oldTable) {
+            while (node != null) {
+                putWithResize(node.key, node.value);
+                node = node.next;
+            }
+        }
+    }
+
+    private void putWithResize(K key, V value) {
+        int index = calculateBucketIndex(key);
+        Node<K, V> currentNode = table[index];
+        if (currentNode == null) {
+            table[index] = new Node<>(key, value, null);
+            size++;
+        } else {
+            Node<K, V> prevNode = null;
+            while (currentNode != null) {
+                if (Objects.equals(currentNode.key, key)) {
+                    currentNode.value = value;
+                    return;
+                }
+                prevNode = currentNode;
+                currentNode = currentNode.next;
+            }
+            prevNode.next = new Node<>(key, value, null);
+            size++;
+        }
+
+        if (size > table.length * DEFAULT_LOAD_FACTOR) {
+            resize();
+        }
+    }
+
+    private int calculateBucketIndex(K key) {
+        return Math.abs(key.hashCode() % table.length);
+    }
+
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private static class Node<K, V> {
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(K key, V value, Node<K, V> next) {
```

Comment:
you may go for private instead? it will be available inside the MyHashMap class anyway